src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveCreateTrueMaskTest.template

```diff
@@ -0,0 +1,268 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Numerics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+namespace JIT.HardwareIntrinsics.Arm
+{
+    public static partial class Program
+    {
+        [Fact]
+        public static void {TestName}()
+        {
+            var test = new CreateTrueMaskTest__{TestName}();
+
+            if (test.IsSupported)
+            {
+                // Validates basic functionality works
+                test.RunBasicScenario_CreateTrueMask();
+
+                // Validates calling via reflection works
+                // TODO-SVE: Enable once register allocation exists for predicates.
+                // test.RunReflectionScenario_CreateTrueMask();
+            }
+            else
+            {
+                // Validates we throw on unsupported hardware
+                test.RunUnsupportedScenario();
+            }
+
+            if (!test.Succeeded)
+            {
+                throw new Exception("One or more scenarios did not complete as expected.");
+            }
+        }
+    }
+
+    public sealed unsafe class CreateTrueMaskTest__{TestName}
+    {
+        private static int FloorPow2(int x)
+        {
+            if (x <= 0)
+            {
+                throw new Exception("Expected a positive argument value.");
+            }
+
+            int result = 1;
+
+            for (x = x / 2; x > 0; x /= 2)
+            {
+                result <<= 1;
+            }
+
+            return result;
+        }
+
+        // DecodePredCount()
+        // =================
+        // integer DecodePredCount(bits(5) bitpattern, integer esize)
```

Comment:
I am not sure if we can have the "copy-pasted" code of this from Arm manual in dotnet codebase because of copyrights. can we simplify the summary docs of `DecodePredCount` instead?

Reply:
Done.